src/AWS.Deploy.CLI/Commands/DeployCommand.cs

```diff
@@ -194,6 +201,129 @@ public async Task ExecuteAsync(string stackName, bool saveCdkProject)
+                        }
+                    }
+                    catch (Exception)
+                    {
+                        throw new InvalidOverrideValueException($"Invalid value {optionSettingValue} for option setting item {optionSettingJsonPath}");
+                    }
+                    
+                    optionSetting.SetValueOverride(settingValue);
+
+                    if (isPartOfDeploymentBundle)
+                        SetDeploymentBundleOptionSetting(recommendation, optionSetting.Id, settingValue);
+
+                    if (optionSetting.Id.Equals("EC2KeyPair"))
+                        await new EC2KeyPairCommand(_toolInteractiveService, _awsResourceQueryer, _consoleUtilities).OverrideValue(recommendation, settingValue.ToString() ?? "");
+                }
+            }
+        }
+
+        private void SetDeploymentBundleOptionSetting(Recommendation recommendation, string optionSettingId, object settingValue)
+        {
+            switch (optionSettingId)
+            {
+                case "DockerExecutionDirectory":
+                    new DockerExecutionDirectoryCommand(_consoleUtilities).OverrideValue(recommendation, settingValue.ToString() ?? "");
+                    break;
+                case "DockerBuildArgs":
+                    new DockerBuildArgsCommand(_consoleUtilities).OverrideValue(recommendation, settingValue.ToString() ?? "");
+                    break;
+                case "DotnetBuildConfiguration":
+                    new DotnetPublishBuildConfigurationCommand(_consoleUtilities).Overridevalue(recommendation, settingValue.ToString() ?? "");
+                    break;
+                case "DotnetPublishArgs":
+                    new DotnetPublishArgsCommand(_consoleUtilities).OverrideValue(recommendation, settingValue.ToString() ?? "");
+                    break;
+                case "SelfContainedBuild":
+                    new DotnetPublishSelfContainedBuildCommand(_consoleUtilities).OverrideValue(recommendation, (bool)settingValue);
+                    break;
+                default:
+                    throw new OptionSettingItemDoesNotExistException($"The Option Setting Item { optionSettingId } does not exist.");
+            }
+        }
+
+        private string GetCloudApplicationName(string? stackName, UserDeploymentSettings? userDeploymentSettings, List<CloudApplication> deployedApplications)
+        {
+            if (!string.IsNullOrEmpty(stackName))
+                return stackName;
+
+            if (userDeploymentSettings == null || string.IsNullOrEmpty(userDeploymentSettings.StackName))
+            {
+                if (_toolInteractiveService.DisableInteractive)
+                {
+                    var message = "The \"--silent\" CLI argument can only be used if a CDK stack name is provided either via CLI argument or through a deployment-settings file. " +

```

Comment:
Be explicit and say which CLI argument to use to set the name. `--stack-name`

Reply:
Fixed